File: 백준/Silver/10773. 제로/제로.py
from collections import deque
import sys

k = int(sys.stdin.readline())
s = deque()

for i in range(k) :
    s.append(int(sys.stdin.readline()))
    if s[-1] == 0 :
        if len(s)==1 :
            break
        else :
            s.pop()
            s.pop()
print(sum(s))

File: 백준/Silver/10773. 제로/제로.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int k = Integer.parseInt(br.readLine());
        Stack<Integer> stack = new Stack<>();
        
        while (k-- > 0) {
        	int n = Integer.parseInt(br.readLine());
        	
        	if (n == 0) {
        		stack.pop();
        	} else {
        		stack.push(n);
        	}
        }
        int size = stack.size();
        int sum = 0;
        
        for (int i = 0; i < size; i++) {
        	sum += stack.pop();
        }
        System.out.println(sum);
    }
}